src/utils/helper.ts

```diff
@@ -17,3 +18,24 @@ export const checkIfUserIsActive = (user: User): boolean => {
   }
   return true
 }
+
+const getIp = async (): Promise<string | undefined> => {
+  try {
+    const data = await axios.get('https://api.ipify.org?format=json')
```

Comment:
Sorry @TahaKhanAbdalli - the clients will be running this on a network without an Internet connection, so we can't make the above call.

Reply:
Then, there's no easy way to fetch the ip with vanilla js

Reply:
Ok. Could you investigate how to do it with node please?  `Soul` supports _extensions_, which are blocks of server-side NodeJS code.  Maybe we can develop a Soul extension to give us the caller's IP address.